docs/usage/self-hosted-configuration.md

```diff
@@ -363,6 +363,10 @@ Before the first commit in a repository, Renovate will:
 The `git` commands are run locally in the cloned repo instead of globally.
 This reduces the chance of unintended consequences with global Git configs on shared systems.
 
+## gitTimeout
```

Comment:
I would like more explanations on:

- when to use the `gitTimeout` config option
- how readers can know if their Git process hangs

Reply:
- When they want the `gitTimeout` to be more than 10 seconds.
- Any git process that uses this config should throw an exception if the timeout exceeds. So logs should be showing this error